### DISCUSSION TITLE
Update password Redirection

### DISCUSSION THREAD

**Discussion by SimsonMoses** [keycloak/keycloak#33855] [Q&A]
Is it possible to redirect direct to update password once the verify email is performed through SPI

       if (event.getType().equals(EventType.VERIFY_EMAIL)) {

            String userId = event.getUserId();
            LOGGER.info("userId : " + userId);
            RealmModel realm = session.getContext().getRealm();
            UserModel user = session.users().getUserById(realm, userId);
            LOGGER.info("User form session: "+user.toString());
            if (user != null) {
                user.addRequiredAction(UserModel.RequiredAction.UPDATE_PASSWORD);
            }
        }
     
        the  update password page has render while submitting with error cookie not found? 
        
        And is it possible to generate the update password token through spi and redirect while verify email event action performed.
        override the URL and redirect to update password page.

        how to achieve the flow? Can anyone help me!!

**Comment by codespearhead** [comment]
Strange. You current code should've worked.

Make sure that code snippet is inside method `public void onEvent(Event event, KeycloakSession session) {...}` and confirm that this issue is still reproducible in the latest stable release of Keycloak, which currently is version `26.0.0`.
